docs/site/hasOne-relation.md

```diff
@@ -0,0 +1,270 @@
+---
+lang: en
+title: 'hasOne Relation'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/hasOne-relation.html
+---
+
+## Overview
+
+A `hasOne` relation denotes a one-to-one connection of a model to another model
+through referential integrity. The referential integrity is enforced by a
+foreign key constraint on the target model which usually references a primary
+key on the source model and a unique constraint on the same column/key to ensure
+one-to-one mapping. This relation indicates that each instance of the declaring
+or source model has exactly one instance of the target model. Let's take an
+example where an application has models supplier and account and a supplier can
+only have one account on the system.
+
```

Comment:
It would be nice to add a diagram like https://guides.rubyonrails.org/association_basics.html#the-has-one-association.